File: Algorithms/Implementation/Designer PDF Viewer.php
<?php

/*
 * Complete the 'designerPdfViewer' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts following parameters:
 *  1. INTEGER_ARRAY h
 *  2. STRING word
 */

function designerPdfViewer($h, $word) {
    $max = 0;
    $width = strlen($word);
    for ($i = 0; $i < $width; $i++) {
        $charIndex = ord($word[$i]) - 97;
        $max = max($max, $h[$charIndex]);
    }
    return $max * $width;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$h_temp = rtrim(fgets(STDIN));
$h = array_map('intval', preg_split('/ /', $h_temp, -1, PREG_SPLIT_NO_EMPTY));
$word = rtrim(fgets(STDIN), "\r\n");
$result = designerPdfViewer($h, $word);

fwrite($fptr, $result . "\n");
fclose($fptr);

File: Algorithms/Implementation/Designer PDF Viewer.py
import os

#
# Complete the 'designerPdfViewer' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER_ARRAY h
#  2. STRING word
#

def designerPdfViewer(h, word):
    # Write your code here
    max = 0
    for i in word:
        if h[ord(i) - 97] > max:
            max = h[ord(i) - 97]
    return max * len(word)


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    h = list(map(int, input().rstrip().split()))
    word = input()
    result = designerPdfViewer(h, word)

    fptr.write(str(result) + '\n')
    fptr.close()
